app/views/home/reference_calculator.haml

```diff
@@ -0,0 +1,44 @@
+- title t('.title')
+
+- content_for :javascript do
+  :javascript
+    $(function() {
+      function update() {
+        var amount = 0;
+        var reference = $('#prefix').val();
+        $('input[data-use-me]').each(function() {
+          var name = $(this).attr('name');
+          var value = Math.round(parseInt($(this).val(), 10));
+          if (value > 0) {
+            amount += value;
+            reference += name + value;
+          }
+        });
+        $('#amount').text(amount);
+        $('#reference').text(reference);
+      }
+
+      $(document).on('input', 'input[data-use-me]', update);
+      update();
+    });
+
+= form_tag finance_create_transaction_collection_path do
+  = hidden_field_tag 'prefix', "FS#{@current_user.ordergroup.id}.#{@current_user.id}"
```

Comment:
this seems pretty hardcoded, would be good to make that more flexible, preferably on some model

Reply:
What about using a helper class for encoding and decoding those strings? I understand the JS part is then a bit of an issue, so either we could compute it server-side, or include a method in the file with the js-computation part (only the computation part, not the display part).

Also, I would really suggest to add a check digit somewhere (perhaps optional, but for generated ones this helps detecting cut-n-paste errors or typos). But can be a feature to add later.

Reply:
i really don't like to calculate it on the server side, but we can but the prefix in the global config. please check out #568 for the parsing part, so we can sync the ideas

>  add a check digit somewhere

it's important that this format can be generated by humans too. they only need to know the number of the ordergroup (which stays the same all the time). As long as the check digit is optional it does not provide a real value and we have a kind of check already, since the sum of the amounts in the reference must match the amount of the bank transaction

Reply:
Ok, fine with the no-check-digit route.

I still think putting the reference computation logic in one place would help in the future, and shouldn't be so involved, right? I don't know, but somehow I'd like to keep this a clearly defined piece, so it can be customized by foodcoops / plugin authors.

Reply:
doing the stuff in a CLEAN way on the server is far from trivial and i don't see the benefits outweigh the effort. If someone will ever implement a plugin, she can update the existing code when needed. i don't want to over engineer it.
maybe you can give #568 a look before we continue here

Reply:
I agree about doing everything server-side, that's too involved.
But what about adding a module in `lib/` that
- converts fields (hash?) to a reference (string)
- converts a reference (string) to fields (hash?)
- emits the javascript needed to convert fields (js object?) to a reference (string)

That would be easy and give flexibility.

Reply:
The js could be something like (just a quick try):

```js
function(prefix, values) {
  var amount = 0;
  var reference = prefix;
  values.each(function(o) {
     var name = o.name;
     var value = Math.round(parseInt(o.value, 10));
     if (value > 0) {
       amount += value;
       reference += name + value;
     }
   });
   return { amount: amount, reference: reference };
}
```

Reply:
i'll give it a try, but i like to get at least an initial review on #568 first, since there might be some overlapping points, which i'd like to implement together

Reply:
One other thing: I'd consider moving the function to an asset, so it can be more easily overridden. Maybe set the function on `window.foodsoftParseBankTransactionReference`? Or make the function name a controller instance variable, so it can be overridden? Deface has issues overriding Javascript in places like this.